File: 1503-last-moment-before-all-ants-fall-out-of-a-plank/1503-last-moment-before-all-ants-fall-out-of-a-plank.cpp
class Solution {
public:
    int getLastMoment(int n, vector<int>& left, vector<int>& right) {
        int leftMax = left.size() ? *max_element(left.begin(), left.end()) : 0;
        int rightMin = right.size() ? *min_element(right.begin(), right.end()) : n;
        return max(leftMax, n - rightMin);
    }
};

File: 1503-last-moment-before-all-ants-fall-out-of-a-plank/1503-last-moment-before-all-ants-fall-out-of-a-plank.java
class Solution {
    public int getLastMoment(int n, int[] left, int[] right) {
        int result = 0;
        for (int ant : left) {
            result = Math.max(result, ant);
        }
        for (int ant : right) {
            result = Math.max(result, n - ant);
        }
        return result;
    }
}